tests/unittests/dataset_tests.py

```diff
@@ -4833,6 +4833,203 @@ def test_select_exclude_dynamic_fields(self):
+        }
+        field_3.info = {"one": {"two": {"three": "test123"}}}
+
+        # doesn't return anything on empty string
+        view = dataset.select_fields(meta_filter="")
+        fields = view.get_field_schema(flat=True)
+        assert "field_1" not in fields
+        assert "field_2" not in fields
+        assert "field_3" not in fields
+        assert "ground_truth" not in fields
+
+        # returns everything on None
+        view = dataset.select_fields(meta_filter=None)
+        fields = view.get_field_schema(flat=True)
+        assert "field_1" not in fields
+        assert "field_2" not in fields
+        assert "field_3" not in fields
+        assert "ground_truth" not in fields
+
+        # basic string match anywhere
+        view = dataset.select_fields(meta_filter="unique")
+        fields = view.get_field_schema(flat=True)
+        assert "field_1" in fields
+        assert "field_2" in fields
+        assert "field_3" not in fields
+        assert "ground_truth" not in fields
+
+        # specific info key: value match with some recursive checking
+        view = dataset.select_fields(meta_filter={"f_3": "oo"})
+        fields = view.get_field_schema(flat=True)
+        assert "field_1" in fields
+        assert "field_2" not in fields
+        assert "field_3" not in fields
+        assert "ground_truth" not in fields
+
+        # should bust the recursion limit (default is 5)
+        view = dataset.select_fields(meta_filter=dict(j_6="nope"))
+        fields = view.get_field_schema(flat=True)
+        assert "field_1" not in fields
+        assert "field_2" not in fields
+        assert "field_3" not in fields
+        assert "ground_truth" not in fields
+
+        # basic string match anywhere
+        view = dataset.select_fields(meta_filter="test123")
+        fields = view.get_field_schema(flat=True)
+        assert "field_1" not in fields
+        assert "field_2" in fields
+        assert "field_3" in fields
+        assert "ground_truth" not in fields
+
+        # match entire info with some additional selected fields
+        view = dataset.select_fields(
+            ["ground_truth", "field_2"],
+            meta_filter=dict(one=dict(two=dict(three="test123"))),
+        )
+        fields = view.get_field_schema(flat=True)
+        assert "field_1" not in fields
+        assert "field_2" in fields
+        assert "field_3" in fields
+        assert "ground_truth" in fields
+
+        # match entire info with no additional selected fields
+        view = dataset.select_fields(
+            meta_filter=dict(one=dict(two=dict(three="test123")))
+        )
+        fields = view.get_field_schema(flat=True)
+        assert "field_1" not in fields
+        assert "field_2" not in fields
+        assert "field_3" in fields
+        assert "ground_truth" not in fields
+
+        # match recursively located key:value
+        view = dataset.select_fields(meta_filter=dict(three="test123"))
```

Comment:
Just confirming: with the current design, there's no way to say you want to match *only* on `description` or on `info`, right?

Reply:
there is now, and also name